include/mbedtls/config.h

```diff
@@ -247,6 +247,34 @@
  */
 //#define MBEDTLS_DEPRECATED_REMOVED
 
+/**
+ * \def MBEDTLS_CHECK_PARAMS
+ *
+ * This configuration controls whether the library validates parameters passed
+ * to it.
+ *
+ * Application code that deals with 3rd party input may wish to enable such
+ * validation, whilst code on closed systems, such as embedded systems, where
+ * the input is controlled and predictable, may wish to disable it entirely to
+ * reduce the code size of the library.
+ *
+ * When the symbol is not defined, no parameter validation except that required
+ * to ensure the integrity or security of the library are performed.
+ *
+ * When the symbol is defined, all parameters will be validated, and an error
+ * code returned where appropriate.
+ */
+#define MBEDTLS_CHECK_PARAMS
+
+/**
+ * \def  MBEDTLS_DEBUG_INVALID_PARAMS
+ *
+ * This configuration controls whether the library will output debug
+ * informattion to stdout if a paramater validation check fails.
+ *
+ */
+//#define MBEDTLS_DEBUG_INVALID_PARAMS
```

Comment:
Do we need this extra public option? Is there another way to avoid this issue with the test? Also, please note that currently the library does not compile with this patch

Reply:
Agreed. It doesn't make enough sense, so I've removed it.